src/Logger.cpp

```diff
@@ -40,39 +41,95 @@ void Logger::Start(time_t time, const StringRef& filename) noexcept
 			lastFlushFileSize = logFile.Length();
 			logFile.Seek(lastFlushFileSize);
 			logFileName.copy(filename.c_str());
-			InternalLogMessage(time, "Event logging started\n");
+			String<40> startMessage;
+			startMessage.printf("Event logging started at level %s\n", logLevel.ToString());
+			InternalLogMessage(time, startMessage.c_str(), MessageLogLevel::INFO);
+			LogFirmwareInfo(time);
 			reprap.StateUpdated();
 		}
 	}
 }
 
+// TODO: Move this to a more sensible location
+void Logger::LogFirmwareInfo(time_t time) noexcept
+{
+	const size_t versionStringLength = 80
+#if 0 && SUPPORT_CAN_EXPANSION // TODO
+			+ reprap.GetExpansion().GetNumExpansionBoards() + 1 * 80
+#endif
+			;
+	char buffer[versionStringLength];
+	StringRef firmwareInfo(buffer, ARRAY_SIZE(buffer));// This is huge but should accomodate for around 20 boards
+	firmwareInfo.printf("Running: %s: %s (%s)", reprap.GetPlatform().GetElectronicsString(), VERSION, DATE);
+
+#if 0 && SUPPORT_CAN_EXPANSION // TODO
+	for (size_t i = 1; i < reprap.GetExpansion().GetNumExpansionBoards() + 1; ++i)
+	{
+		auto expansionBoardData = reprap.GetExpansion().FindIndexedBoard(i);
+		firmwareInfo.catf(" - %s", expansionBoardData.typeName);
+	}
+#endif
+	InternalLogMessage(time, firmwareInfo.c_str(), MessageLogLevel::INFO);
+}
+
 void Logger::Stop(time_t time) noexcept
 {
 	if (logFile.IsLive() && !inLogger)
 	{
 		Lock loggerLock(inLogger);
-		InternalLogMessage(time, "Event logging stopped\n");
+		InternalLogMessage(time, "Event logging stopped\n", MessageLogLevel::INFO);
 		logFile.Close();
 		reprap.StateUpdated();
 	}
 }
 
-void Logger::LogMessage(time_t time, const char *message) noexcept
+// This will not start the logger if it is currently stopped
+void Logger::SetLogLevel(LogLevel newLogLevel) noexcept
 {
-	if (logFile.IsLive() && !inLogger)
+	if (logLevel != newLogLevel)
+	{
+		logLevel = newLogLevel;
+		reprap.StateUpdated();
+	}
+}
+
+//bool Logger::IsLoggingEnabledFor(const MessageType mt) const noexcept
```

Comment:
Don't leave commented out functions. If you think a function might be needed later, use: #if 1 <comment that it may be needed later> #else <the unused code> #endif

Reply:
Added comments.